src/api/tx_initiation/builder/transaction_details_builder.rs

```diff
@@ -82,6 +85,20 @@ impl TransactionDetailsBuilder {
         self
     }
 
+    /// Add many public announcements.
+    ///
+    /// Use this method for public announcements that are *not* encrypted UTXO
+    /// notifications. The encrypted UTXO notifications are generated on the fly
+    /// at a later stage.
+    pub fn public_announcements<Iter: IntoIterator<Item = PublicAnnouncement>>(
+        mut self,
+        public_announcements: Iter,
+    ) -> Self {
+        self.public_announcements
+            .append(&mut public_announcements.into_iter().collect_vec());
```

Comment:
```suggestion
self.public_announcements.extend(public_announcements)
```

* more concise & idiomatic
* more efficient: avoids intermediate allocation